comment/tests/test_service.py

```diff
@@ -98,18 +103,24 @@ def test_send_messages(self):
             self.email_service.get_message('test subject', 'test body', ['test@test'], html_msg='test msg')
             for _ in range(100)
         ]
+
         self.assertEqual(len(mail.outbox), 0)
+
         self.email_service.send_messages(messages)
+
         self.assertNotEqual(len(mail.outbox), 100)
         self.assertTrue(self.email_service._email_thread.is_alive())
         self.email_service._email_thread.join()
+
         self.assertEqual(len(mail.outbox), 100)
 
     def test_send_confirmation_request_django(self):
```

Comment:
this test along with [`test_send_confirmation_request_django`](https://github.com/Radi85/Comment/pull/189/files#diff-f2e7d9c153f774cd79acae65f97ea94af7b74f04760f3d9e37dec05bbfec5ccaR132) is flaky at the moment(at least on my machine). I'll try to investigate further about this.

Reply:
I am not sure that I understood your point.

Reply:
I am not sure that I understood your point.

Reply:
if you run these tests a couple of times, they seem to fail sometimes. At least, that was the behaviour on my end.

I will try to make a video and post it, if it can't be reproduced at your level. Maybe when I can have some more peace of mind.

Reply:
I feel the issue might be related to `len(mail.outbox)`. However, this never happened at my end neither in travis.

Reply:
no, it is actually the key send to the email that is causing some weird issues.

In the video below, I have run the module `test_service` every 3 seconds and tried to highlight the changes in output. As you may see below, it fails sometimes.

![comment-flaky-tests](https://user-images.githubusercontent.com/13880786/114869527-b64d8400-9e14-11eb-9663-0318a4a0b738.gif)

Reply:
Can you share the traceback? It's not clear in the record.

Reply:
```sh
FAIL: test_send_confirmation_request_django (comment.tests.test_service.TestDABEmailService)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/mock.py", line 1325, in patched
    return func(*newargs, **newkeywargs)
  File "Comment/comment/tests/test_service.py", line 126, in test_send_confirmation_request_django
    self.email_contents_test(
  File "Comment/comment/tests/test_service.py", line 36, in email_contents_test
    self.assertIs(arg in contents, True, msg=f'{arg} not present in email contents')
AssertionError: False is not True : /comment/confirm/.eJxtjk0KgzAQha8i2VZlJmqhWfUmJZihBDITMclCSu_esXTZ7fve38u0Qrtx0lLqzZqlklTjzJZLjfLsvGQ5OLfSrZn5ZL0h9jGpp1KpA979yjRSaErOFAVFFiwOMA-4dGjdfHVoxxtMEy4XAAegXr9tD_FMvzFVOAdKf7Wonaj1fv--O8--P_IYOsM:1lX1Ns:c1ipHKhTIFkoAd823EnP1wQfj0Ph6wdBs0Uw1lWlqfI/ not present in email contents
```

this is when one of the test fails.

Reply:
```sh

FAIL: test_send_confirmation_request_api (comment.tests.test_service.TestDABEmailService)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/mock.py", line 1325, in patched
    return func(*newargs, **newkeywargs)
  File "Comment/comment/tests/test_service.py", line 140, in test_send_confirmation_request_api
    self.email_contents_test(
  File "Comment/comment/tests/test_service.py", line 36, in email_contents_test
    self.assertIs(arg in contents, True, msg=f'{arg} not present in email contents')
AssertionError: False is not True : /api/comments/confirm/.eJxtjk0KwyAQha8ibpuEUWIhrnqTImYogjOGqItSeveOpctu3_f-XrpXPLXnnvOkY-GG3LTXR6kt8UMFLvyk0quKhWiwSSOFlMXTsLbZ3EIkXHDvQkYKd0EWrJlhnY1Txvp1825dNuvAXC8AHkC84TjuHAh_Y6JQ2TH_1ZJ0GqkP5_fdOPv-APWaOs4:1lX1RS:ex75U6gCasy3ooESdwhGVYxKYgyCfSXnjP6y7_fKZ-Y/ not present in email contents

======================================================================
FAIL: test_send_confirmation_request_django (comment.tests.test_service.TestDABEmailService)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/mock.py", line 1325, in patched
    return func(*newargs, **newkeywargs)
  File "Comment/comment/tests/test_service.py", line 126, in test_send_confirmation_request_django
    self.email_contents_test(
  File "Comment/comment/tests/test_service.py", line 36, in email_contents_test
    self.assertIs(arg in contents, True, msg=f'{arg} not present in email contents')
AssertionError: False is not True : /comment/confirm/.eJxtjk0KwyAQha8ibpuEUWIhrnqTImYogjOGqItSeveOpctu3_f-XrpXPLXnnvOkY-GG3LTXR6kt8UMFLvyk0quKhWiwSSOFlMXTsLbZ3EIkXHDvQkYKd0EWrJlhnY1Txvp1825dNuvAXC8AHkC84TjuHAh_Y6JQ2TH_1ZJ0GqkP5_fdOPv-APWaOs4:1lX1RS:ex75U6gCasy3ooESdwhGVYxKYgyCfSXnjP6y7_fKZ-Y/ not present in email contents

----------------------------------------------------------------------
Ran 15 tests in 1.188s

```

when both of them fail.